projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts

```diff
@@ -0,0 +1,41 @@
+import { AppConfigService } from '@alfresco/adf-core';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AcaMobileAppSwitcherService {
+  constructor(private config: AppConfigService) {}
+
+  checkForMobileApp() {
+    const mobileAppSwitchConfig: any = this.config.get<any>('mobileAppSwitch', {});
+    const ua = navigator.userAgent.toLowerCase();
+    const isAndroid = ua.indexOf('android') > -1;
+    const isIphone = ua.indexOf('iphone') > -1;
+    if (isIphone === true) {
+      const url = window.location.href;
+      const app = {
+        launchApp: () => {
+          window.location.href = mobileAppSwitchConfig.isIphone + url;
+        },
+        openWebApp: () => {
+          window.location.href = mobileAppSwitchConfig.appleStore;
+        }
+      };
+      app.launchApp();
+    } else if (isAndroid === true) {
+      const url = window.location.href;
+      const app = {
+        launchApp: () => {
+          setTimeout(() => {
+            window.location.href = mobileAppSwitchConfig.isAndroidPart1 + url + mobileAppSwitchConfig.isAndroidPart2;
+          }, 500);
+        },
+        openWebApp: () => {
+          window.location.href = mobileAppSwitchConfig.googlePlayStore;
+        }
+      };
+      app.launchApp();
```

Comment:
do you need creating an "app" object just to call "app.launchApp()"? why now running that directly?